training/eval/bucketing.py

```diff
@@ -0,0 +1,231 @@
+from typing import List, Tuple
+from sklearn.preprocessing import MinMaxScaler
+from eval.eval_utils import add_labeler_id_column
+import pandas as pd
+import numpy as np
+import ast
+
+
+def get_buckets_bounds_and_scaler(
+    dataset_df: pd.DataFrame,
+    tr_preds_df: pd.DataFrame,
+    metric: str,
+) -> Tuple[List[float], MinMaxScaler]:
+    """
+    Returns the bounds for the buckets for the given metric.
+    Args:
+        dataset_df: DataFrame as loaded from newsroom csv. Should contain
+            columns for requested metric (metric itself with all ratings and
+            split), `ArticleID` and `System`.
+        tr_preds_df: DataFrame with predictions for the training split. Should
+            contain columns `ArticleID`, `System` and `deberta_reward`.
```

Comment:
I would add assert if we require some specific columns:
`assert set(tr_preds_df.columns).issuperset({"ArticleID", "System", "deberta_reward"})`